pandas/core/construction.py

```diff
@@ -611,6 +611,8 @@ def sanitize_array(
         if hasattr(data, "__array__"):
             # e.g. dask array GH#38645
             data = np.asarray(data)
+            if copy:
```

Comment:
Is there an efficient way to check if asarray already made a copy?

Reply:
Maybe just change the call above to `data = np.array(data, copy=copy)`?

Reply:
Thx, good point